lib/util/memcache.ex

```diff
@@ -0,0 +1,60 @@
+# Copyright(c) 2015-2018 ACCESS CO., LTD. All rights reserved.
+
+use Croma
+
+defmodule Antikythera.Memcache do
+  @moduledoc """
+  In-memory cache for each executor pool.
+
+      iex> Antikythera.Memcache.write("foo", "bar", epool_id)
+      :ok
+      iex> Antikythera.Memcache.read("foo", epool_id)
+      {:ok, "bar"}
+  """
+
+  @default_lifetime 1800
+  @default_ratio    0.9
+
+  alias Croma.Result, as: R
+  alias Antikythera.Time
+  alias AntikytheraCore.ExecutorPool.MemcacheManager
+  alias Antikythera.ExecutorPool.Id, as: EPoolId
+  alias AntikytheraCore.Ets.Memcache
+  alias Antikythera.Memcache.NormalizedFloat
+
+  defun read(key :: term, epool_id :: v[EPoolId.t]) :: R.t(term, :not_found) do
+    case Memcache.read(key, epool_id) do
+      {:ok, {_, expire_at, prob_expire_at, value}} ->
+        if is_expired(expire_at, prob_expire_at) do
+          {:error, :not_found}
+        else
+          {:ok, value}
+        end
+      {:error, :not_found} -> {:error, :not_found}
+    end
```

Comment:
Let's use `R.bind/2`:
```ex
Memcache.read(key, epool_id)
|> R.bind(fn {_, expire_at, prob_expire_at, value} ->
  if is_expired(expire_at, prob_expire_at) do
    {:error, :not_found}
  else
    {:ok, value}
  end
end)
```

Reply:
in 282e91f79e0ee0aa32c5c1a5ea680e34b89a665e